app/test_main.py

```
@@ -1,3 +1,43 @@
+import pytest
 from app.main import get_human_age
 
-# write your code here
+
+@pytest.mark.parametrize(
+    "cat_in, dog_in, expected_cat, expected_dog",
+    [
+        (2, 3, 0, 0),
+        (14, 14, 0, 0),
+        (15, 15, 1, 1),
+        (16, 16, 1, 1),
+        (23, 23, 1, 1),
+        (24, 24, 2, 2),
+        (27, 27, 2, 2),
+        (28, 28, 3, 2),
+        (29, 29, 3, 3),
+        (33, 33, 4, 3),
+        (37, 37, 5, 4),
+        (41, 41, 6, 5),
+        (45, 45, 7, 6),
+        (49, 49, 8, 7),
+        (53, 53, 9, 7),
+        (57, 57, 10, 8),
+        (61, 61, 11, 9),
+        (65, 65, 12, 10),
+        (100, 100, 21, 17),
+    ],
+)
+def test_get_human_age(
+    cat_in: int,
+    dog_in: int,
+    expected_cat: int,
+    expected_dog: int
+) -> None:
+    result = get_human_age(cat_in, dog_in)
+    assert result[0] == expected_cat, (
+        f"Wiek Kota: Oczekiwano {expected_cat}, otrzymano {result[0]} "
+        f"dla wejścia {cat_in}"
+    )
+    assert result[1] == expected_dog, (
+        f"Wiek Psa: Oczekiwano {expected_dog}, otrzymano {result[1]} "
+        f"dla wejścia {dog_in}"
+    )
```

This test covers the main logic well, but you're still missing tests for crucial edge cases. You'll need to add one or more new test functions to verify the function's behavior with:

1.  **Negative numbers**: What should `get_human_age(-5, -10)` return?
2.  **Incorrect data types**: The function should raise an exception (like `TypeError`) for non-integer inputs. You can test this using `pytest.raises`.

This violates checklist item #2: 'Make sure you've tests checks all edged situations... such as negative numbers... or if the function receives an incorrect type of data'.